tests/lib/gui/lvgl/src/main.c

```diff
@@ -71,7 +71,9 @@ ZTEST(lvgl_screen, test_add_delete_screen)
 
 	lv_scr_load(new_screen);
 
+#ifndef CONFIG_SOC_FAMILY_STM32
```

Comment:
Can you explain why this is needed?

Reply:
I just ran the test on my mimxrt1060_evkb and the issue is also present on there. From my point of view there are two possibilities:

- Find out what the issue is and possibly submit upstream patch (I suspect that something is going wrong in the animation code.
- Remove the handler call completely. If we are sure that `lv_img_set_src` will call through to the decoder/fs layer even without the handler I am fine with removing the handler. For the screen creation we are not using an animation, therefore they should not be needed.

Reply:
I am inclined to follow the second route.

Reply:
@ajarmouni-st 
Today I tried fiddeling with various `STACK_SIZE` settings but I could not pin point the issue further. Stacks that I checked:
```
CONFIG_ISR_STACK_SIZE=4096
CONFIG_INPUT_THREAD_STACK_SIZE=4096
CONFIG_ZTEST_STACK_SIZE=4096
CONFIG_LV_Z_FLUSH_THREAD=n
```
I also tried increasing `CONFIG_LV_Z_MEM_POOL_SIZE`, but that did not matter either. Need to do some further tests tomorrow.

Reply:
Hi @faxe1008 thanks for taking the time to investigate this. I also tried to increase all kind of stack sizes but it had no effect whatsoever.

Reply:
I verified that the decoder is accessing the fs implementation even without the call to the handler:
```
#0  lvgl_fs_read (drv=0x80a2a00 <fs_drv>, file=0x80a3a64 <lvgl_heap_mem+3684>, buf=0xf7c782b4, btr=4, br=0xf7c781e4) at /zephyr/modules/lvgl/lvgl_fs.c:99
#1  0x0807e752 in lv_fs_read (file_p=0xf7c78230, buf=0xf7c782b4, btr=4, br=0xf7c7822c) at /modules/lib/gui/lvgl/src/misc/lv_fs.c:218
#2  0x08065197 in lv_img_decoder_built_in_info (decoder=0x80a2c6c <lvgl_heap_mem+108>, src=0x8097b39, header=0xf7c782b4)
    at /modules/lib/gui/lvgl/src/draw/lv_img_decoder.c:294
#3  0x08065bd0 in lv_img_decoder_get_info (src=0x8097b39, header=0xf7c782b4) at /modules/lib/gui/lvgl/src/draw/lv_img_decoder.c:99
#4  0x08086741 in lv_img_set_src (obj=0x80a3a04 <lvgl_heap_mem+3588>, src=0x8097b39) at /modules/lib/gui/lvgl/src/widgets/lv_img.c:107
```
So removing this one should be ok - however I am intruiged to find out why this happens 😄

Reply:
> I verified that the decoder is accessing the fs implementation even without the call to the handler

Great, Thanks!

> however I am intruiged to find out why this happens

Me too, but it appears to be really convoluted (I spent a considerable time investigating this before finding that commenting the handler calls solves the issue, & that's when I gave up ^ ^)